packages/purpose-process/test/getPurposeById.test.ts

```diff
@@ -301,6 +295,148 @@ describe("getPurposeById", () => {
     });
   });
 
+  it("should get the purpose with the risk analysis form if the requester is an e-service delegated producer", async () => {
+    const producer = {
+      ...getMockTenant(),
+      kind: tenantKind.PA,
+    };
+
+    const producerDelegate = {
+      ...getMockTenant(),
+      kind: tenantKind.PA,
+    };
+
+    const mockEService: EService = {
+      ...getMockEService(),
+      producerId: producer.id,
+    };
+    const mockPurpose1: Purpose = {
+      ...getMockPurpose(),
+      eserviceId: mockEService.id,
+      riskAnalysisForm: getMockValidRiskAnalysisForm(tenantKind.PA),
+      delegationId: generateId<DelegationId>(),
+    };
+
+    const consumerDelegation = getMockDelegation({
+      id: mockPurpose1.delegationId,
+      kind: delegationKind.delegatedConsumer,
+      eserviceId: mockPurpose1.eserviceId,
+      delegatorId: mockPurpose1.consumerId,
+      state: delegationState.active,
+    });
+
+    const producerDelegation = getMockDelegation({
+      kind: delegationKind.delegatedProducer,
+      eserviceId: mockPurpose1.eserviceId,
+      delegatorId: producer.id,
+      delegateId: producerDelegate.id,
+      state: delegationState.active,
+    });
+
+    await addOnePurpose(mockPurpose1);
+    await addOneDelegation(consumerDelegation);
+    await addOneDelegation(producerDelegation);
+    await addSomeRandomDelegations(mockPurpose1, addOneDelegation);
+    await addOneEService(mockEService);
+    await addOneTenant(producer);
+    await addOneTenant(producerDelegate);
+
+    const result = await purposeService.getPurposeById(
+      mockPurpose1.id,
+      producerDelegate.id,
+      genericLogger
+    );
+    expect(result).toMatchObject({
+      purpose: mockPurpose1,
+      isRiskAnalysisValid: true,
+    });
+  });
+
+  it("should get the purpose with the risk analysis form if the requester is an e-service producer", async () => {
+    const producer = {
+      ...getMockTenant(),
+      kind: tenantKind.PA,
+    };
+
+    const mockEService: EService = {
+      ...getMockEService(),
+      producerId: producer.id,
+    };
+    const mockPurpose1: Purpose = {
+      ...getMockPurpose(),
+      eserviceId: mockEService.id,
+      riskAnalysisForm: getMockValidRiskAnalysisForm(tenantKind.PA),
+      delegationId: generateId<DelegationId>(),
+    };
+
+    const consumerDelegation = getMockDelegation({
+      id: mockPurpose1.delegationId,
+      kind: delegationKind.delegatedConsumer,
+      eserviceId: mockPurpose1.eserviceId,
+      delegatorId: mockPurpose1.consumerId,
+      delegateId: producer.id,
```

Comment:
This should not be the producer otherwise we are bypassing the case we want to test

```suggestion
      delegateId: generateId<TenantId>(),
```